sdk/core/Azure.Core/src/Shared/MessagingClientDiagnostics.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using Azure.Core.Pipeline;
+
+#nullable enable
+
+namespace Azure.Core.Shared
+{
+    /// <summary>
+    /// Client Diagnostics support for messaging clients. Currently, this is only used for AMQP clients.
+    /// HTTP libraries should use the ClientDiagnostics type instead.
+    /// </summary>
+    internal class MessagingClientDiagnostics
+    {
+        private readonly string _fullyQualifiedNamespace;
```

Comment:
Is `fullyQualifiedNamespace` a bit too "EventHubs/ServiceBus" for a generic thinger?

Reply:
I think since this is currently only for AMQP it is okay. If/when we extend this for REST-based libraries we can revisit.